[Brandon Thio] iP

beetee17 opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/19).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

beetee17 committed:
Implement a skeletal version of Duke that starts by greeting the user, simply echos commands entered by the user, and exits when the user types 'bye'

beetee17 committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

beetee17 committed:
Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

beetee17 committed:
Add support for tracking three types of tasks: todo, deadline and event

beetee17 committed:
Handle various errors such as incorrect inputs entered by the user.

beetee17 committed:
Add support for deleting tasks from the list.

beetee17 committed:
Use the input/output redirection technique to semi-automate the testing of Duke.

beetee17 committed:
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when Duke starts up.

beetee17 committed:
Store deadline dates as a java.time.LocalDate in task objects. Accept dates in a format such as yyyy-mm-dd format (e.g., 2019-10-15) and print in a different format such as MMM dd yyyy e.g., (Oct 15 2019).

beetee17 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Task.java

beetee17 committed:
Fix unable to parse date when it has surrounding whitespace

beetee17 committed:
Organise the classes into suitable java packages.

beetee17 committed:
Extract the following classes:
Ui: deals with interactions with the user
Storage: deals with loading tasks from the file and saving tasks in the file
Parser: deals with making sense of the user command
TaskList: contains the task list e.g., it has operations to add/delete tasks in the list
Command classes (i.e., AddCommand, DeleteCommand, ExitCommand etc.) that inherit from an abstract Command class,

beetee17 committed:
Rename delete function

beetee17 committed:
Use Gradle to automate some of the build tasks of the project.

beetee17 committed:
Package the app as an executable JAR file so that it can be distributed easily.

beetee17 committed:
Setup dummy JUnit tests

beetee17 committed:
Organise commands, parser, storage, ui into their own packages

beetee17 committed:
Add JavaDoc comments to the code.

beetee17 committed:
Tweak the code to comply with a coding standard

beetee17 committed:
Give users a way to find a task by searching for a keyword.

beetee17 committed:
Merge branch 'branch-A-CodingStandard'

beetee17 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/commands/ListTasksCommand.java

beetee17 committed:
Add JUnit tests to test the behavior of the code.

beetee17 committed:
Use checkStyle to detect coding style violations.

beetee17 committed:
Got javaFX minimal example working

jeromepui commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,40 @@
+package duke.task;
+import duke.exceptions.*;
```
Imported classes should always be listed explicitly.
```suggestion
import duke.exceptions.ParsingTaskException;
```

jeromepui commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+public class Task {
+    private static final String DATA_PATH = new File("").getAbsolutePath() + "/data/duke.txt";
+    private String title;
+    private boolean completed;
```
I can tell from your code that the boolean tracks whether the task is completed. Perhaps you should rename it to sound more like a boolean.
```suggestion
    private boolean isCompleted;
```

jeromepui commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+ * Deadlines are tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
`by` seems vague for a variable name. Perhaps you could rename it to make it clearer.

beetee17 committed:
Add minimal GUI using JavaFX

jeromepui commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        List<Task> tasks = taskList.getAll().stream()
```
Good use of Java pipelines to find tasks! 👍

jeromepui reviewed (commented):
LGTM. 👍 Just some minor fixes to be made.

adeearyaa commented on src/main/java/duke/commands/ExitCommand.java:
```diff
@@ -0,0 +1,15 @@
+public class ExitCommand extends Command {
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        ui.showGoodByeMessage();
```
good descriptive method names

adeearyaa commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
```
Great use of streams!

adeearyaa commented on src/main/java/duke/exceptions/InvalidDeadlineException.java:
```diff
@@ -0,0 +1,10 @@
+public class InvalidDeadlineException extends InvalidCommandException {
+    public InvalidDeadlineException() {
+        super("Could not parse deadline. To create a deadline, "
+                + "please use the format in this example: "
```
nice exception message

adeearyaa commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,76 @@
+
+import duke.commands.*;
+import duke.enums.Action;
+import duke.exceptions.*;
```
could have specified which file to import instead of all maybe?

adeearyaa reviewed (commented):
Great code!

beetee17 committed:
Merge branch 'branch-A-Checkstyle'

# Conflicts:
#	src/main/java/duke/commands/FindCommand.java
#	src/main/java/duke/commands/ListTasksCommand.java
#	src/main/java/duke/ui/Ui.java

beetee17 committed:
Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code

beetee17 committed:
Remove dead code and replace if-else branching with switch statement for readability.

beetee17 committed:
Merge pull request #1 from beetee17/branch-A-CodeQuality

Replace if-else branching with switch statement for readability

beetee17 committed:
Merge pull request #2 from beetee17/branch-A-Assertions

Use assert feature (not JUnit assertions) to document important assumptions

beetee17 committed:
Provide a way to tag items

For example, users can now tag a task as "fun" via the command
"tag <index> #fun".

Current limitation is that the feature does not allow users to add
multiple tags at once. For example, it does not allow the command
"tag <index> #urgent, #important, #school".

beetee17 committed:
Add a User Guide to the project

beetee17 committed:
Move Readme to docs folder

beetee17 committed:
Update the main Readme page

beetee17 committed:
Release the product to be used by potential users
